tests/vpp.cpp

```diff
@@ -125,27 +135,53 @@ class VppTest
         return bool(m_allocator);
     }
 
+    void flushGPU()
+    {
+        std::deque<SharedPtr<VideoFrame> > frames;
+        SharedPtr<VideoFrame> f;
+        do {
+            f = m_allocator->alloc();
+            if (!f)
```

Comment:
Thanks for suggestion @MicheleL-Intel, 
But I don't think we should deal with the fail of alloc(), because:
1, alloc() just require a surface from pool, if the pool is  empty, alloc() return NULL, else a frame.

> SharedPtr<VideoFrame> PooledFrameAllocator::alloc()
{
    return  m_pool->alloc();
}

2, m_allocator surface pool's surface elements are created by setformat(). When setformat() failed to create surface from driver, it return false to up layer functions.

> bool PooledFrameAllocator::setFormat(uint32_t fourcc, int width, int height)
{
    std::vector<VASurfaceID> surfaces;
    surfaces.resize(m_poolsize);
......
    VAStatus status = vaCreateSurfaces(*m_display, VA_RT_FORMAT_YUV420, width, height,
            &surfaces[0], surfaces.size(),
            &attrib, 1);

    if (status != VA_STATUS_SUCCESS) {
        ERROR("create surface failed, %s", vaErrorStr(status));
        return false;
    }